src/embedding/ad.ts

```diff
@@ -0,0 +1,584 @@
+// (C) 2020 GoodData Corporation
+
+import isEmpty = require('lodash/isEmpty');
+import {
+    CommandFailed,
+    IObjectMeta,
+    IGdcMessageEvent,
+    IGdcMessageEventDataWrapper,
+    ICommandFailedBody
+} from './common';
+
+type ANALYTICAL_DESIGNER = 'analyticalDesigner';
+const AD_PRODUCT_NAME = 'analyticalDesigner';
+
+// TODO: these are part of client (gooddata-js), move to model (typings)
+export interface IBaseExportConfig {
+    title?: string;
+    format?: 'xlsx' | 'csv' | 'raw';
+    mergeHeaders?: boolean;
+}
+
+// TODO: these are part of client (gooddata-js), move to model (typings)
+// NOTE: AFM is ommitted on purpose; it should be added by AD itself; create new type using Omit<>
+export interface IExportConfig extends IBaseExportConfig {
+    showFilters?: boolean;
+}
+
+export namespace EmbeddedAnalyticalDesigner {
+    export type IADMessageEvent<T, TBody> = IGdcMessageEvent<ANALYTICAL_DESIGNER, T, TBody>;
+
+    export enum GdcAdCommandType {
+        Save = 'adSave',
+        SaveAs = 'adSaveAs',
+        Export = 'adExport',
+        Undo = 'adUndo',
+        Redo = 'adRedo'
+    }
+
+    export enum GdcAdEventType {
+        NewInsightInitialized = 'adNewInsightInitialized',
+        InsightOpened = 'adInsightOpened',
+        InsightSaved = 'adInsightSaved',
+        UndoFinished = 'adUndoFinished',
+        RedoFinished = 'adRedoFinished',
+        ExportFinished = 'adExportFinished'
+    }
```

Comment:
Since we remove `ad` incomming command, we should remove it here as well
```
export enum GdcAdEventType {
        NewInsightInitialized = 'newInsightInitialized',
        InsightOpened = 'insightOpened',
        InsightSaved = 'insightSaved',
        UndoFinished = 'undoFinished',
        RedoFinished = 'redoFinished',
        ExportFinished = 'exportFinished'
    }
```

Reply:
Camel case please

Reply:
updated